grafana/grafana
# Influxdb Datasource: Remove angular dependencies for Influxdb influxql annotations

Fixes https://github.com/grafana/grafana/issues/50637

**What this PR does / why we need it**: This PR removes angular dependencies for InfluxDB, creates annotations with a React component using the Influx Query Language(InfluxQL) and migrates the legacy annotations. Angular will not be supported in Grafana 10.

**Special notes for your reviewer**:
To test this PR, do the following:

1. Check out main.
2. Create a dashboard with an InfluxDB InfluxQL datasource(two provisioned examples are given with query examples that return different data, but if you have a custom InfluxDB datasource, please share or mention that in your review)
```make devenv sources=influxdb,influxdv1```
  - gdev-influxdb-influxql
      SELECT * from cpu
  - gdev-influxdb1-influxql
      SELECT * from cpu
4. Create an annotation for each. Below are some examples.
  - gdev-influxdb-influxql
      SELECT * from logs where time >= now() - 60m and time <= now()
  - gdev-influxdb1-influxql
      SELECT mean("value") FROM "logins.count" WHERE time >= now() - 20m and time <= now()
6. Save the Dashboard.
7. Check that the annotation is displayed on the dashboard.
8. Checkout this branch `bohandley/influxdb-annotations-react`
9. Check that the previous annotation is still displayed.
10. Create a new annotation and check that it is displayed on this branch and see that it is displayed.
